python/TestHarness/testers/Tester.py

```diff
@@ -397,6 +402,18 @@ def getRedirectedOutputFiles(self, options):
         """ return a list of redirected output """
         return [os.path.join(self.getTestDir(), self.name() + '.processor.{}'.format(p)) for p in xrange(self.getProcs(options))]
 
+    def addCaveats(self, caveats):
+        """ Add caveat(s) which will be displayed with the final test status """
+        if type(caveats) == type([]):
```

Comment:
Not a huge fan of this line. How much word would it be to just make sure that caveats is always a list?